### ISSUE TITLE
[GSOC 2021] Starter issues

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#10195]
As mentioned on our [GSoC organization page](https://summerofcode.withgoogle.com/organizations/6623198768005120/), we require participants to have submitted at least 2 non-trivial, substantially different mypy or mypyc PRs.

We'll collect ideas about suitable issues here. You are free to work on any other open issues as well or file new issues if you have ideas (but wait for feedback from project maintainers before starting work on a new issue).

* [Mypy issues tagged as "good first issue"](https://github.com/python/mypy/issues?q=is%3Aopen+is%3Aissue+label%3Agood-first-issue)
* [Mypyc issues tagged as "good first issue"](https://github.com/mypyc/mypyc/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)

mypyc/mypyc#644 is a good, relevant meta-issue with a lot of bite-sized tasks.

Here are some high impact "good second issues" suggested by @hauntsaninja:
* #9880
* #8481 (comment)
* #9852

You can also work on multiple issues tagged as "good first issue".

Feel free to leave specific questions in individual issues (that are relevant to the issue). You can leave more general questions here or in our [Gitter chat](https://gitter.im/mypyc-dev/community).

If you have more ideas, please leave suggestions in comments.

Various helpful links:

* [Quick start to contributing mypy](https://github.com/python/mypy#quick-start-for-contributing-to-mypy)
* [How to contribute to mypy](https://github.com/python/mypy/blob/master/CONTRIBUTING.md)
* [Mypyc GitHub project and readme](https://github.com/mypyc/mypyc)
* [Mypyc user documentation](https://mypyc.readthedocs.io/en/latest/)
* [Mypyc developer documentation](https://github.com/python/mypy/blob/master/mypyc/doc/dev-intro.md)
* [test-data/unit/README.md](https://github.com/python/mypy/blob/master/test-data/unit/README.md): How to verify your changes locally
* [Python C API documentation](https://docs.python.org/3/extending/extending.html)
* [Python C API reference](https://docs.python.org/3/c-api/index.html)
* [@TH3CHARLie's blog post about GSoC 2020](https://mypy-lang.blogspot.com/2021/03/summer-of-code-with-mypy.html)

**Comment by JukkaL** [comment]
mypyc/mypyc#757 is another good second issue. It contains many ideas, some of which are challenging, but you can just pick one idea and focus on it.

**Comment by Sangu80** [comment]
@JukkaL how can i get a chance to work on to your organization projects , like is there any tasks that i need to solve to get selected for gsoc under your organization

**Comment by hauntsaninja** [comment]
Closing, since it's late 2023 and most of the specific issue links are completed
